### PR TITLE
MicroProfile Telemetry 2.0

### PR DESCRIPTION
**Pull request by radcortez**

### CODE REVIEW

File: testsuite/tck/src/test/java/io/smallrye/opentelemetry/tck/TestConfigSource.java [hunk 1]
```diff
@@ -7,9 +7,9 @@
 public class TestConfigSource extends MapBackedConfigSource {
     public TestConfigSource() {
         super("TestConfigSource",
-                Map.of("otel.sdk.disabled", "true",
-                        "otel.traces.exporter", "none",
-                        "otel.metrics.exporter", "none"),
+                Map.of("otel.traces.exporter", "none",
+                        "otel.metrics.exporter", "none",
+                        "otel.logs.exporter", "none"),
```
Comment (brunobat) [inline testsuite/tck/src/test/java/io/smallrye/opentelemetry/tck/TestConfigSource.java:12]: They say in the TCK that the exporter must be set to logging

**Comment by brunobat** [comment]
@radcortez Seems that we need to set `otel.metric.export.interval=3000` to make the runs reproducible.
Does this fix the flaky tests?

**Comment by radcortez** [comment]
The Flaky tests are all in the JVM package: https://github.com/eclipse/microprofile-telemetry/tree/877f460eb1c3675102b2b5038cb48dec6d99bf92/tck/metrics/src/main/java/org/eclipse/microprofile/telemetry/metrics/tck/jvm

I haven't looked into the details yet. I suspect that the metrics collection for JVM attributes just takes longer than the configured export interval.

**Comment by brunobat** [comment]
> The Flaky tests are all in the JVM package: https://github.com/eclipse/microprofile-telemetry/tree/877f460eb1c3675102b2b5038cb48dec6d99bf92/tck/metrics/src/main/java/org/eclipse/microprofile/telemetry/metrics/tck/jvm
> 
> I haven't looked into the details yet. I suspect that the metrics collection for JVM attributes just takes longer than the configured export interval.

CC @Emily-Jiang

**Comment by radcortez** [comment]
The problem is that the number of metrics generated cycles the log file, so there may be occasions where the metrics collected are not in the file anymore. The log file cannot be cycled.

**Comment by Emily-Jiang** [comment]
@radcortez @brunobat can you try out [this](https://github.com/eclipse/microprofile-telemetry/pull/222) PR to see whether it fixes your issue?

**Comment by Emily-Jiang** [comment]
I have restaged the Telemetry 2.0 release [here](https://oss.sonatype.org/content/repositories/orgeclipsemicroprofile-1708) with the above fix. Please try it out @brunobat @radcortez

**Comment by radcortez** [comment]
It consistently fails in `JvmGarbageCollectionTest` with JDK 11 (confirmed locally). However, it does work fine with JDK 17 and 21. We use the implementation provided by OpenTelemetry.

It seems that with JDK 11, there is not a single GC during the test execution, so the metric is not generated. Apparently it does work by hinting a `System.gc()` in the test method.

**Review by brunobat — DISMISSED** [review]

**Pull request merged**